File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int element = -1, counter = 0;
        
        for (auto x : nums){
            if (element == x){
                counter++;
            }
            else if (counter > 0){
                counter--;
            }
            else {
                element = x;
                counter = 1;
            }
        }
        return element;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] arr) {
        int ans = -1, counter = 0;
        int n = arr.length;
        for (int i = 0; i < n; i++){
            if (counter == 0 && ans != arr[i]){
                ans = arr[i];
                counter++;
            }
            else if(ans == arr[i]) {
                counter++;
            }
            else {
                counter--;
            }
        }
        return ans;
    }
}